rocky/tools/forms/boefje.py

```diff
@@ -0,0 +1,67 @@
+from django import forms
+from django.utils.translation import gettext_lazy as _
+
+from octopoes.models.types import ALL_TYPES
+from tools.enums import SCAN_LEVEL
+from tools.forms.base import BaseRockyForm
+from tools.forms.settings import (
+    BOEFJE_CONSUMES_HELP_TEXT,
+    BOEFJE_DESCRIPTION_HELP_TEXT,
+    BOEFJE_PRODUCES_HELP_TEXT,
+    BOEFJE_SCAN_LEVEL_HELP_TEXT,
+    BOEFJE_SCHEMA_HELP_TEXT,
+)
+
+OOI_TYPE_CHOICES = sorted((ooi_type.get_object_type(), ooi_type.get_object_type()) for ooi_type in ALL_TYPES)
+
+
+class BoefjeAddForm(BaseRockyForm):
+    oci_image = forms.CharField(
+        required=True,
+        label=_("Container image"),
+        widget=forms.TextInput(
+            attrs={
+                "description": "The name of the Docker image. For example: ghcr.io/minvws/openkat/nmap",
+                "aria-describedby": "input-description",
+            }
+        ),
+    )
+    name = forms.CharField(
+        required=True,
+        label=_("Name"),
+    )
+    description = forms.CharField(
+        required=False,
+        label=_("Description"),
+        widget=forms.Textarea(attrs={"rows": 3}),
+        help_text=BOEFJE_DESCRIPTION_HELP_TEXT,
+    )
+    oci_arguments = forms.CharField(
+        required=False,
+        label=_("Arguments"),
+        widget=forms.TextInput(
+            attrs={"description": "For example: -sTU --top-ports 1000", "aria-describedby": "input-description"}
+        ),
+    )
+    schema = forms.JSONField(
+        required=False,
+        label=_("JSON Schema"),
+        help_text=BOEFJE_SCHEMA_HELP_TEXT,
+    )
+    consumes = forms.CharField(
+        required=False,
+        label=_("Input object type"),
+        widget=forms.Select(choices=OOI_TYPE_CHOICES),
+        help_text=BOEFJE_CONSUMES_HELP_TEXT,
+    )
+    produces = forms.CharField(
+        required=False,
+        label=_("Output mime types"),
```

Comment:
I think we should perhaps mention that if people want normalizers of existing boefjes to trigger from their new boefje, they need to add `boefje/{boefje-id}` here for the normalizers that take in `boefje-id` and need to trigger. For instance, to trigger the `nmap` normalizer we need `boefje/nmap` here at least.

Reply:
Do you think it would be enough to add this in the help text?

Reply:
Perhaps, although it's perhaps easy to overlook. @underdarknl thinking of this, is this what we want? Since I used the `produces` field for this but now realize that this will indeed add that mime-type to every output of the new boefje. It does seem the easiest way to create a custom nmap/dns-sec at this point and especially when copying a boefje in the future.

Reply:
The listed mime-types are just that, listed. they dont actually do anything. Its up to the boefje itself to provide the actual mime-type for its output, and in that i can mimic another (already existing boefje) by using the boefje/$boefjename mimetype, or it can target any of the normalizer directly by using the normalizer/$normalizername mimetype.